app/src/main/java/org/mozilla/tv/firefox/experiments/FretboardProvider.kt

```diff
@@ -31,13 +31,25 @@ class FretboardProvider(private val applicationContext: Context) : CoroutineScop
     private val experimentSource = KintoExperimentSource(
             EXPERIMENTS_BASE_URL, EXPERIMENTS_BUCKET_NAME, EXPERIMENTS_COLLECTION_NAME
     )
+
+    // We use the telemetry clientId because the data team wants to be able to reproduce
+    // buckets server-side in order to verify that clients are being bucketed correctly.
+    // This is low risk because:
+    // a) ID only gets sent to Mozilla servers
+    // b) ID + bucket doesn't leak personally identifiable information
     val fretboard: Fretboard = Fretboard(experimentSource, FlatFileExperimentStorage(experimentsFile),
             object : ValuesProvider() {
                 override fun getClientId(context: Context): String {
                     return TelemetryIntegration.INSTANCE.clientId
                 }
             })
 
+    // A list of the experiment names with the user's branch (A or B) appended.
+    // If the user is in the experiment (active), they are in branch B, otherwise branch A.
+    private val experimentBranches: List<String>
+        get() = fretboard.getExperimentsMap(applicationContext)
+                .map { it.key + if (it.value) ":B" else ":A" }
```

Comment:
Also for future thought - I just talked to @suyounghong about how the Data team handles naming conventions for experiments, and he said in Normandy (Desktop experiments) usually there's a "tag" for the experiment, a bug number to uniquely cross-ref, and a "branch", so the whole experiment name might look like "onboarding_bug123456_b", which seems like something we should emulate for whatever our backend does.

Reply:
to elaborate on this. we handle the experiment and the branch as separate fields in telemetry. so, for example, if we have 2 branches in experiment-XXXX (where XXXX is the number of the bugzilla bug that launched the experiment), then one profile would have: 

```
{"experiment_id": "experiment-XXXX", "branch": "control"}
```
whereas another profile would have: 
```
{"experiment_id": "experiment-XXXX", "branch": "variant1"}
```

if it is the case that we can only use one field, and needed to combine the experiment_id and the branch, I would recommend using a unique deliminator like: 

```
{"experiment_id_and_branch": "experiment-XXXX:control"}
```

Reply:
Just the active experiments is fine. 

I'd like to keep the naming convention as close to desktop as possible.  Currently focus has something like `{"experiment-name":"branch"}` but having a bug reference in the name as well is useful. It might be a good idea to check in with @colintheshots as to how he implemented this in Focus.

Reply:
I interpreted 'active' to mean whether you are in the experiment. So I thought that `getExperimentsMap` gets both branches - those in the experiment, B, and those not, A. But `getActiveExperiments` would only get the users that are in branch B.

How do you run an experiment with more than two options? My understanding was that you specify one range of buckets, so you are either in the experiment or you are not.

I could be understanding things completely wrong - @colintheshots would you mind verifying/clarifying?

Reply:
@liuche I think 'active' experiment means the user is in that experiment as per `fretboard#getActiveExperiments` implementation: 

`fun getActiveExperiments(context: Context): List<Experiment> { return experiments.filter { isInExperiment(context, ExperimentDescriptor(it.name)) }}`

Reply:
Right, I think that makes sense that we'd want to collect "active experiments". Since we're also creating the experiments ourselves, we know what *other* experiments this user isn't in. With every release, a user is being shown some version of an experiment, even if that is the control. So we'd still be able to measure usage by comparing users who are in "onboarding-a" and "onboarding-b" where either could be the control.

I guess I don't understand why we'd need to collect the experiments that a users **isn't** in?

Reply:
Doesn't not being in the experiment mean they are in branch A? So if we only get active experiments, that would always be branch B, so then we would never tag the control branch.

Reply:
There's a difference between active experiments and experiment branches. A user can be actively in an experiment but on the control branch (no treatment).  We need a list of the active experiments that a user is in as well as which branch (A, B, C...) they are in for each active experiment.

Reply:
Not all users will necessarily be in every experiment. Most experiments will only be active for some subset of the total population. Within that subset, the users will be assigned to a particular branch. So we need to have both the experiment id and the branch name. Those not in the experiment will not be tagged with the active experiment or a branch.

Reply:
Interesting! Okay, my only question that @dnarcese made me think of with this approach is, if the users not in an experiment don't have any tag (e.g., control for experimentX), how do you draw conclusions comparing the two populations (e.g. experiment outperformed control) when the control group for an experiment doesn't have a tag or something distinguishing them from users who, say, are stuck on an older version and also don't have any experiment tags? Can you filter the data based on app release version (e.g., v4.2+ only because that's when we release experiment X) or other ways of distinguishing?
Thanks for all the context @mmccorks! If just collecting activeExperiments is enough, let's do that - I just wanted to double check to make sure my understanding was correct.

Reply:
The two populations we compare are the two branches - the control group will be tagged as one of the branches. We need them explicitly tagged as part of the experiment to be able to draw conclusions like you said - with the correct version etc. so that we are only comparing people explicitly in the experiment. People in the control are IN the experiment, even if they are not receiving a treatment.

Reply:
We need the active experiments, and within those active experiments, which branch the user is on in that experiment.